engine/lib/filestore.php

```diff
@@ -487,6 +487,10 @@ function _elgg_filestore_init() {
 
 	// Unit testing
 	elgg_register_plugin_hook_handler('unit_test', 'system', '_elgg_filestore_test');
+
+	// Move entity icons if entity owner has changed
+	elgg_register_event_handler('update:after', 'object', '_elgg_filestore_move_icons');
```

Comment:
After some contemplation I decided there is room for this in core. I am sure plugins forget to do this all the time. File plugin doesn't do this but probably should (haven't actually tested changing the owner).